Branch-SDK/Branch-SDK/BranchOpenRequest.m

```diff
@@ -98,16 +98,16 @@ - (void)processResponse:(BNCServerResponse *)response error:(NSError *)error {
     // Update session params
     preferenceHelper.sessionParams = sessionData;
     
-    // If referable, also se tup install params
-    if (preferenceHelper.isReferrable) {
-        // If present, set it.
-        if (sessionData) {
-            preferenceHelper.installParams = sessionData;
-        }
-        // If not present, only allow nil to be set if desired (don't clear otherwise)
-        else if (self.allowInstallParamsToBeCleared) {
-            preferenceHelper.installParams = nil;
-        }
+    // Scenarios:
+    // If isReferrable is false, don't set, period.
+    // Otherwise, if isReferrable, is from a link click, and
+    // * Install: set to whatever we get.
+    // * Open and installParams set: don't set.
+    // * Open and not installParams and isReferrable: set if not null.
+    BOOL dataIsFromALinkClick = [sessionData rangeOfString:@"\"+clicked_branch_link\":0"].location == NSNotFound;
```

Comment:
Can we be guaranteed that there won't be spaces or anything between ":0? Why not just parse the string and look for the specific key?